google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/TableAdminClient.java

```diff
@@ -0,0 +1,759 @@
+   * @return ConsistencyToken
+   */
+  public ConsistencyToken generateConsistencyToken(String tableId) {
+    return TableAdminResponses.convertTokenResponse(
+        this.stub
+            .generateConsistencyTokenCallable()
+            .call(composeGenerateConsistencyTokenRequest(tableId)));
+  }
+
+  /**
+   * Generates a token to verify the replication status of table mutations invoked before this call
+   * asynchronously Token expires in 90 days
+   *
+   * <p>Sample code:
+   *
+   * <pre><code>
+   *  try(TableAdminClient client =  TableAdminClient.create(InstanceName.of("[PROJECT]", "[INSTANCE]"))) {
+   *        client.generateConsistencyToken("tableId");
+   *   }
+   * </code></pre>
+   *
+   * @param tableId
+   * @return ApiFuture<ConsistencyToken>
+   */
+  public ApiFuture<ConsistencyToken> generateConsistencyTokenAsync(String tableId) {
+    ApiFuture<GenerateConsistencyTokenResponse> tokenResp =
+        this.stub
+            .generateConsistencyTokenCallable()
+            .futureCall(composeGenerateConsistencyTokenRequest(tableId));
+
+    return ApiFutures.transform(
+        tokenResp,
+        new ApiFunction<GenerateConsistencyTokenResponse, ConsistencyToken>() {
+          @Override
+          public ConsistencyToken apply(GenerateConsistencyTokenResponse input) {
+            return TableAdminResponses.convertTokenResponse(input);
+          }
+        });
+  }
+
+  /**
+   * Checks replication consistency for the specified token consistency token
+   *
+   * <p>Sample code:
+   *
+   * <pre><code>
+   *  try(TableAdminClient client =  TableAdminClient.create(InstanceName.of("[PROJECT]", "[INSTANCE]"))) {
+   *        client.isConsistent("tableId", token);
+   *   }
+   * </code></pre>
+   *
+   * @param tableId
+   * @param token
+   * @return boolean
+   */
+  public boolean isConsistent(String tableId, ConsistencyToken token) {
+    return stub.checkConsistencyCallable()
+        .call(token.toProto(getTableName(tableId)))
+        .getConsistent();
+  }
+
+  /**
+   * Checks replication consistency for the specified token consistency token asynchronously
+   *
+   * <p>Sample code:
+   *
+   * <pre><code>
+   *  try(TableAdminClient client =  TableAdminClient.create(InstanceName.of("[PROJECT]", "[INSTANCE]"))) {
+   *        client.isConsistentAsync("tableId", token);
+   *   }
+   * </code></pre>
+   *
+   * @param tableId
+   * @param token
+   * @return ApiFuture<Boolean>
+   */
+  public ApiFuture<Boolean> isConsistentAsync(String tableId, ConsistencyToken token) {
+    ApiFuture<CheckConsistencyResponse> CheckConsResp =
+        stub.checkConsistencyCallable().futureCall(token.toProto(getTableName(tableId)));
+
+    return ApiFutures.transform(
+        CheckConsResp,
+        new ApiFunction<CheckConsistencyResponse, Boolean>() {
+          @Override
+          public Boolean apply(CheckConsistencyResponse input) {
+            return input.getConsistent();
+          }
+        });
+  }
```

Comment:
Please add `ApiFuture<Void> awaitConsistency()`

Reply:
I am not clear on the use-case. Let's cover this on the next call.